File: CinemaReservations/Java/CinemaAllocationsDomain/src/main/java/com/sdd/cinemaallocations/NoPossibleAdjacentSeatsFound.java
package com.sdd.cinemaallocations;

import lombok.EqualsAndHashCode;
import lombok.ToString;
import lombok.Value;

import java.util.List;

@Value
@ToString(callSuper = true)
@EqualsAndHashCode(callSuper = true)
public class NoPossibleAdjacentSeatsFound extends SeatsAllocated {

    public NoPossibleAdjacentSeatsFound(int partyRequested, List<Seat> reservedSeats) {
        super(partyRequested, reservedSeats);
    }
}

File: CinemaReservations/CSharp/CinemaAllocations.Domain/NoPossibleAdjacentSeatsFound.cs
using System.Collections.Generic;

namespace CinemaAllocations.Domain
{
    public class NoPossibleAdjacentSeatsFound : SeatsAllocated
    {
        public NoPossibleAdjacentSeatsFound(int partyRequested) : base(new List<Seat>(), partyRequested)
        {
        }
    }
}